holmscan/scan.py

```diff
@@ -16,73 +15,80 @@ class Scan(HolmscanModule):
     def __init__(self, controller):
         super(Scan, self).__init__(controller)
 
-    @login
-    def get_web_scan(self, id):
-        url = u"{0}scan/wruns/{1}".format(self.controller.conf.get("HOLMSEC_URL"), id)
-
-        response = self.controller.session.get(url).text
-
-        return response
-
-    @login
-    def start_web_scan(self, **kwargs):
-        data = kwargs
-        data["assets"] = kwargs["assets"]
-        data.update(
-            {
-                "appliance": "-1",
-                "source": 0,
-                "node": "",
-                "user": "{0}".format(self.user_id),
-                "start_date": None,
-            }
-        )
-
-        job_url = u"{0}scan/wscans/".format(self.controller.conf.get("HOLMSEC_URL"))
-        job_response = self.controller.session.post(
-            job_url,
-            data=json.dumps(data).replace(" ", ""),
-            headers={"Content-Type": "application/json"},
-        )
+    def get_net_assets(self):
+        url = "{}/net-scans/assets".format(self.controller.conf.get("HOLMSEC_ENDPOINT"))
+        headers = {
+            "Authorization": "Token {}".format(
+                self.controller.conf.get("HOLMSEC_TOKEN")
+            )
+        }
+        response = self.controller.session.get(url, headers=headers)
```

Comment:
Having to build up the authorization header for every call outside the real call is redundant and prone to errors. Move the headers build into the `.get()` call. If the .get call is the requests session object, then consider to make a wrapper call in the self.controller class to expose a .get() call that constructs the header and just passes it along to the session.get() method call. To preserve the feature of having the ability to add more header fields, a kwargs could be accepted in the new .get() method that would merge with the authorization header into a single dict and then passed to requests session.

Reply:
Will look into this. Stay tuned.

Reply:
I have now moved `.get()` and `.post()` into the controller object.